mmv1/third_party/terraform/services/resourcemanager/data_source_google_organizations.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package resourcemanager
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+func DataSourceGoogleOrganizations() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceGoogleOrganizationsRead,
+		Schema: map[string]*schema.Schema{
+			"filter": {
+				Optional: true,
+				Type:     schema.TypeString,
+			},
+			"organizations": {
+				Computed: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"directory_customer_id": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"display_name": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"lifecycle_state": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"name": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"org_id": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func datasourceGoogleOrganizationsRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	organizations := make([]map[string]interface{}, 0)
+
+	filter := ""
+	if v, ok := d.GetOk("filter"); ok {
+		filter = v.(string)
+	}
+
+	request := config.NewResourceManagerClient(userAgent).Organizations.Search(&cloudresourcemanager.SearchOrganizationsRequest{
+		Filter: filter,
+	})
+
+	err = request.Pages(context.Background(), func(organizationList *cloudresourcemanager.SearchOrganizationsResponse) error {
+		for _, organization := range organizationList.Organizations {
+			organizations = append(organizations, map[string]interface{}{
+				"directory_customer_id": organization.Owner.DirectoryCustomerId,
```

Comment:
The API docs say that `Owner` is required, so it should never be `nil` here but for safety's sake we should check for `nil` before dereferencing it.

Reply:
Hi @SirGitsalot,

no worries.
The better the review, the better the result.

I've just added the improvement and I hope I understood you right:
https://github.com/GoogleCloudPlatform/magic-modules/compare/c6da0f5e69dcfc471d3f114283edb989aad984c8..885b464bb967cc41cec87755283e17960c7e3969

Happy weekend!